File: companymicroservice/src/main/java/arctic/tn/mohamedalirepo/entity/Subscription.java
package arctic.tn.mohamedalirepo.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.*;

import javax.persistence.*;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.Date;

@Entity
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Subscription implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Temporal(TemporalType.DATE)
    private Date dateDebut;
    @Temporal(TemporalType.DATE)
    private Date dateFin;
    private Float prixAbon;
    @Enumerated(EnumType.STRING)
    private TypeSubscription typeSubscription;
    @ManyToOne

    private Entreprise entreprise;
}

File: markmyflow-frontEnd/src/app/core/models/subscription.ts
import { Company } from "./company";
import { TypeSubscription } from "./type-subscription";

export class Subscription {
    id: number;
    prixAbon: number;
    dateDebut: Date;
    dateFin: Date;
    typeSubscription: TypeSubscription;
    entreprise?: Company;
  
    constructor() {
      this.id = 0;
      this.prixAbon = 0.0; // declare prix_abon as a float variable
      this.dateDebut = new Date();
      this.dateFin = new Date();
      this.typeSubscription = TypeSubscription.annuel;
      this.entreprise = new Company();
    }
  }
